### PR TITLE
cmd: implement abidump

### PR DESCRIPTION
**Pull request by holiman**
This PR implements `abidump`, a utility to parse calldata against the fourbyte directory database. 
```
[user@work abidump]$ ./abidump 
Error: one argument needed
Usage: ./abidump <hexdata>

Parses the given ABI data and tries to interpret it from the fourbyte database.
```

```
[user@work abidump]$ ./abidump a9
WARNING: Transaction data is not valid ABI (missing the 4 byte call prefix)
```

```
[user@work abidump]$ ./abidump a9059cbb
WARNING: Transaction contains data, but provided ABI signature could not be verified: signature "function transfer(address , uint256 ) returns()" matches, but arguments mismatch: abi: cannot marshal in to go type: length insufficient 0 require 32
```
```
[user@work abidump]$ ./abidump a9059cbb000000000000000000000000ea0e2dc7d65a50e77fc7e84bff3fd2a9e781ff5c0000000000000000000000000000000000000000000000015af1d78b58c40000
Info: transfer(address: 0xEA0e2Dc7d65A50E77FC7E84bff3FD2A9E781ff5c,uint256: 25000000000000000000)
```

### CODE REVIEW

**Comment by holiman** [comment]
@karalabe you requested this once upon a time, over discord... Want it?

**Comment by holiman** [comment]
Rebased, let's see what travis thinks of it now

**Comment by holiman** [comment]
peter

File: cmd/abidump/main.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package main
```
Comment (karalabe) [inline cmd/abidump/main.go]: (c) header :P

File: cmd/abidump/main.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The go-ethereum Authors
```
Comment (gballet) [inline cmd/abidump/main.go]: it's 2020 now

Comment (holiman) [reply cmd/abidump/main.go]: but it wasn't when I wrote the code

Comment (gballet) [reply cmd/abidump/main.go]: you didn't specify a license when you wrote the code :grin:

**Pull request merged**